### ISSUE TITLE
Terraform Error: File Not Found

### ISSUE THREAD

**Issue by seancallaway** [serverless-ca/terraform-aws-ca#100]
**Module Version:** 1.0.1
**Terraform Version:** 1.5.7
**AWS Provider Version:** 5.26.0

When running `terraform apply`, I am getting the following errors:

```
Error: local-exec provisioner error

  with module.certificate_authority.module.rsa_issuing_ca_crl_lambda.null_resource.install_python_dependencies,
  on .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/main.tf line 13, in resource "null_resource" "install_python_dependencies":
  13:   provisioner "local-exec" {

Error running command 'chmod +x .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh ./.terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh
': exit status 127. Output: /bin/sh:
./.terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh: not found
```

This repeats a few times.

Obviously, I can see [that script](https://github.com/serverless-ca/terraform-aws-ca/blob/main/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh) and my [Terraform output](https://paste.ofcode.org/LS6vjgua7v2spe95Lhty2t) seems to show that the module downloads correctly. I'm not sure what's going on here.

My usage of the module is as follows:

```
module "certificate_authority" {
  source  = "serverless-ca/ca/aws"
  version = "1.0.1"
  project = "[REDACTED]"
  env     = "[REDACTED]"

  root_ca_info    = var.root_ca_info
  issuing_ca_info = var.issuing_ca_info
  cert_info_files = ["tls", "revoked", "revoked-root-ca"]
  providers       = {
    aws           = aws
    aws.us-east-1 = aws.us-east-1
  }
}
```

The variables used are defined as follows:

```
variable "root_ca_info" {
  description = "Information used for the creation of the Root CA"
  default     = {
    country              = "US"
    locality             = "[REDACTED]"
    lifetime             = 7300
    organization         = "[REDACTED]"
    organizationalUnit   = "[REDACTED]"
    commonName           = "[REDACTED]"
    pathLengthConstraint = 1
  }
}

variable "issuing_ca_info" {
  description = "Information used for the creation of the Issuing CA"
  default     = {
    country              = "US"
    locality             = "[REDACTED]"
    lifetime             = 3650
    organization         = "[REDACTED]"
    organizationalUnit   = "[REDACTED]"
    commonName           = "[REDACTED]"
    pathLengthConstraint = 0
  }
}
```

**Comment by paulschwarzenberger** [comment]
I suspect this is a local issue with your laptop - what operating system are you using?
As a test, I just did a local Terraform apply on MacOS for the [Cloud CA repository](https://github.com/serverless-ca/cloud-ca). Here's an excerpt from the logs:
`module.certificate_authority.module.rsa_tls_cert_lambda.null_resource.install_python_dependencies: Provisioning with 'local-exec'...
module.certificate_authority.module.create_rsa_root_ca_lambda.null_resource.install_python_dependencies (local-exec): Executing: ["/bin/sh" "-c" "chmod +x .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh\n./.terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh\n"]
module.certificate_authority.module.create_rsa_issuing_ca_lambda.null_resource.install_python_dependencies (local-exec): Executing: ["/bin/sh" "-c" "chmod +x .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh\n./.terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh\n"]
module.certificate_authority.module.rsa_tls_cert_lambda.null_resource.install_python_dependencies (local-exec): Executing: ["/bin/sh" "-c" "chmod +x .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh\n./.terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh\n"]
module.certificate_authority.aws_s3_object.cert_info["tls"]: Modifying... [id=tls.json]
module.certificate_authority.aws_s3_object.csrs["cloud-architect.csr"]: Modifying... [id=csrs/cloud-architect.csr]
module.certificate_authority.aws_s3_object.cert_info["revoked"]: Modifying... [id=revoked.json]`

**Comment by paulschwarzenberger** [comment]
If you use GitHub, it might be worth deploying using GitHub Actions, rather than locally, using the Cloud CA repository workflows as an example.

**Comment by paulschwarzenberger** [comment]
I think there may be an issue with [this code](https://github.com/serverless-ca/terraform-aws-ca/blob/main/modules/terraform-aws-ca-lambda/main.tf) on some operating systems:
```
provisioner "local-exec" {
    command = <<-EOT
      chmod +x ${path.module}/scripts/lambda-build/create-package.sh
      ./${path.module}/scripts/lambda-build/create-package.sh
    EOT
```
I believe it should be:
```
provisioner "local-exec" {
    command = <<-EOT
      chmod +x ${path.module}/scripts/lambda-build/create-package.sh
      ${path.module}/scripts/lambda-build/create-package.sh
    EOT
```

**Comment by seancallaway** [comment]
Thanks for #101. Will test it shortly.

This is all running in GitLab CI, so it's running in an Aline Linux container (`hashicorp/terraform:1.5.7`).

**Comment by paulschwarzenberger** [comment]
Module upgraded to `v1.1.0`, please let me know if this fixes the issue.

**Comment by seancallaway** [comment]
Still seeing the error using 1.1.0.

```
Downloading registry.terraform.io/serverless-ca/ca/aws 1.1.0 for certificate_authority...
...
module.certificate_authority.module.rsa_root_ca_crl_lambda.null_resource.install_python_dependencies: Creating...
module.certificate_authority.module.rsa_root_ca_crl_lambda.null_resource.install_python_dependencies: Provisioning with 'local-exec'...
module.certificate_authority.module.rsa_root_ca_crl_lambda.null_resource.install_python_dependencies (local-exec):Executing: ["/bin/sh" "-c" "chmod +x .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh\n.terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh\n"]
module.certificate_authority.module.create_rsa_root_ca_lambda.null_resource.install_python_dependencies (local-exec):/bin/sh: .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh: not found
module.certificate_authority.module.create_rsa_issuing_ca_lambda.null_resource.install_python_dependencies (local-exec):/bin/sh: .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh: not found
module.certificate_authority.module.rsa_root_ca_crl_lambda.null_resource.install_python_dependencies (local-exec):/bin/sh: .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh: not found

Error: local-exec provisioner error

   with module.certificate_authority.module.rsa_root_ca_crl_lambda.null_resource.install_python_dependencies,
 on .terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/main.tf line 13, in resource "null_resource" "install_python_dependencies":
 13:   provisioner "local-exec" {

Error running command 'chmod +x
.terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh
.terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh
': exit status 127. Output: /bin/sh:
.terraform/modules/certificate_authority/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh: not found
```

This repeats again for `create_rsa_issuing_ca_lambda`, `rsa_issuing_ca_crl_lambda`, `rsa_tls_cert_lambda`, and `create_rsa_root_ca_lambda`.

Would be more than happy to test branches, if you'd like.

**Comment by seancallaway** [comment]
> Alpine Linux

That's what Hashicorp uses to build the [Terraform Container](https://github.com/hashicorp/terraform/blob/main/Dockerfile).

**Comment by paulschwarzenberger** [comment]
I suspect this may be a difference with how `chmod` works on Alpine Linux compared with Ubuntu / MacOS

**Comment by seancallaway** [comment]
That's possible, but I haven't seen an issue with it before.

Here's a test in a container using `chmod +x` on a file and it behaves how I'd expect.

![image](https://github.com/serverless-ca/terraform-aws-ca/assets/1229384/6dfb82c3-4e61-4aa3-95a2-1219819275af)

Would it make sense for the scripts to have their execution bit turned on in the Git repo and eliminate the need for the `chmod` via `local-exec`?

**Comment by paulschwarzenberger** [comment]
@seancallaway the [script](https://github.com/serverless-ca/terraform-aws-ca/blob/main/modules/terraform-aws-ca-lambda/scripts/lambda-build/create-package.sh) begins with `#!/bin/bash` to specify the interpreter. If bash is not present in that location, it could result in the file not found error.

Can you please try installing bash in your Alpine Linux container to test this, see [this Stack Overflow article](https://stackoverflow.com/questions/40944479/docker-how-to-use-bash-with-an-alpine-based-docker-image)

**Comment by seancallaway** [comment]
Oooh, a BASH shebang would likely be a problem, though it's interesting that it appears on the chmod line.

I'll investigate.

**Comment by seancallaway** [comment]
So, I've gotten the script to work. Since there was nothing BASH-specific in the script, I changed the shebang to /bin/sh, since that's available on just about every Unix-like OS, including Ubuntu, Mac, and Alpine. However, the script then failed because I had Python 3.11 installed. While working through that, I noted that `virtualenv` was being used instead of `venv`.

I've opened a [pull request](https://github.com/serverless-ca/terraform-aws-ca/pull/102) to resolve these issues.
